infra/datalake/datalake_stack.py

```diff
@@ -9,10 +9,10 @@ class DataLakeStack(core.Stack):
     """Data Lake stack definition."""
 
     # pylint: disable=redefined-builtin,too-many-locals
-    def __init__(self, scope: core.Construct, id: str, **kwargs) -> None:
+    def __init__(self, scope: core.Construct, id: str, deploy_env, **kwargs) -> None:
         super().__init__(scope, id, **kwargs)
 
-        ENV = self.stack_name.split("-")[-1]
+        ENV = deploy_env
```

Comment:
We should not override "constants" within methods. That basically makes them globals. We might as well refer to `deploy_env` within this method.

Reply:
Fixed in f62ceb21ca87043e481e3eff95beadabd334daba